### PR TITLE
[SPARK-7146][ML] Expose the common params as a DeveloperAPI for other ML developers

### PR DESCRIPTION
**Pull request by holdenk**
## What changes were proposed in this pull request?

Expose the common params from Spark ML as a Developer API.

## How was this patch tested?

Existing tests.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala [hunk 1]
```diff
@@ -177,9 +177,11 @@ private[shared] object SharedParamsCodeGen {
 
     s"""
       |/**
-      | * Trait for shared param $name$defaultValueDoc.
+      | * Trait for shared param $name$defaultValueDoc. These traits may be changed or
```
Comment (MLnick) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: Should probably be "This trait"?

Comment (holdenk) [inline mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala]: Good point.

**Comment by holdenk** [comment]
cc @jkbradley

**Comment by holdenk** [comment]
And updated after being called out in a talk and Spark Summit :p

**Comment by holdenk** [comment]
Jenkins retest this please

**Comment by holdenk** [comment]
Merged into master

**Pull request closed without merging**